presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -257,7 +266,20 @@ private Date getDate(int columnIndex, DateTimeZone localTimeZone)
         }
 
         try {
-            return new Date(DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value)));
+            long millis = DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value));
+            if (millis >= START_OF_MODERN_ERA) {
+                return new Date(millis);
+            }
+
+            // The chronology used by default by Joda is not historically accurate for dates preceding the introduction of the Gregorian calendar
```

Comment:
This is a bit long, which makes it hard to read. Since we have multiple lines of text, wrapping at ~100 would look better and be easier to read.